### PR TITLE
[QUIC] Removes abstract providers and mock from S.N.Quic

### LINKED ISSUE #54050: [QUIC] HTTP/3 AltSvc Mock tests disabled
`HttpClientHandler_AltSvc_Test` tests for mock quic implementation and for upgrades from HTTP/2 fail regularly in CI with:
```System.TimeoutException : The operation has timed out.```

### PR DESCRIPTION
**Pull request by ManickaP**
Guts out mock provider and the abstract machinery in S.N.Quic enabling to have multiple implementation.
~Leaves in one static `QuicProvider` with `IsSupported`, `CreateConnectionAsync` and `CreateListenerAsync`.~
Also removes `QuicProvider` and moves statics into `QuicListener` and `QuicConnection` as discussed in API Review.

This change is not changing anything functionally (for MsQuic implementation), only removes (Mock implementation) and shuffles some code around.

Further step: move code from `MsQuic...` classes directly to `Quic...` classes. It'll make debugging easier but it'll make future experiments with different implementation slightly harder. It still might be worth it, but left out of this PR for now.

I've checked the generated assemblies and `IsSupported` doesn't throw anywhere as is expected.

Fixes #54050, #53688, #55995, #70424, #56005

### CODE REVIEW

File: src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicPlatformDetectionTests.cs [hunk 1]
```diff
@@ -17,8 +16,8 @@ public class MsQuicPlatformDetectionTests : QuicTestBase<MsQuicProviderFactory>
         [ConditionalFact(nameof(IsQuicUnsupported))]
         public void UnsupportedPlatforms_ThrowsPlatformNotSupportedException()
         {
-            Assert.Throws<PlatformNotSupportedException>(() => CreateQuicListener());
-            Assert.Throws<PlatformNotSupportedException>(() => CreateQuicConnection(new IPEndPoint(IPAddress.Loopback, 0)));
+            Assert.ThrowsAsync<PlatformNotSupportedException>(async () => await CreateQuicListener());
+            Assert.ThrowsAsync<PlatformNotSupportedException>(async () => await CreateQuicConnection(new IPEndPoint(IPAddress.Loopback, 0)));
```
Comment (rzikm) [inline src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicPlatformDetectionTests.cs:20]: This should still work, right?

```suggestion
            Assert.ThrowsAsync<PlatformNotSupportedException>(() => CreateQuicListener());
            Assert.ThrowsAsync<PlatformNotSupportedException>(() => CreateQuicConnection(new IPEndPoint(IPAddress.Loopback, 0)));
```

**Review by rzikm — APPROVED** [review]
LGTM!

File: src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicPlatformDetectionTests.cs (hunk 1 above)
Comment (ManickaP) [reply src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicPlatformDetectionTests.cs:20]: It's `ValueTask` vs `Task` problem. So either async await combo, or `AsTask`. I don't care which way.

File: src/libraries/Common/tests/System/Net/Http/Http3LoopbackServer.cs [hunk 2]
```diff
@@ -43,7 +42,7 @@ public Http3LoopbackServer(QuicImplementationProvider quicImplementationProvider
                 MaxBidirectionalStreams = options.MaxBidirectionalStreams,
             };
 
-            _listener = new QuicListener(quicImplementationProvider ?? QuicImplementationProviders.Default, listenerOptions);
+            _listener = QuicListener.ListenAsync(listenerOptions).Result;
```
[the remark this answers is not in the document]
Comment (ManickaP) [reply src/libraries/Common/tests/System/Net/Http/Http3LoopbackServer.cs:45]: When it becomes truly async in the future, we should probably move it out of the ctor.
I added `Debug.Assert` that should suffice for now.

**Comment by ManickaP** [comment]
@JamesNK you'll need to react to this change in Kestrel. Relevant diff is the ref source: https://github.com/dotnet/runtime/pull/70421/files#diff-7c92d08d67d2432a1047dbb476f1388567c182213fd2b801c14cdb84d59ee359
Removed are provider classes, `IsSupported` is moved to `QuicListener` (or `QuicConnection`) and creation methods are now static and named: `QuicListener.ListendAsync(options)` and `QuicConnection.ConnectAsync(options)`.

**Pull request merged**